### ISSUE TITLE
[FR] Python bindings

### ISSUE THREAD

**Issue by chr1sj0nes** [google/benchmark#954]
We have created some simple Python bindings for google/benchmark that we would like to contribute.

The intent is to allow the following usage:
```python
import benchmark

@benchmark.register       # Equivalent to `BENCHMARK(f)` in C++.
def bench_index(state):   # state is a shim for `benchmark::State`.
  l = list(range(10000))  # Setup code is not benchmarked (as with C++).
  while state:            # Equivalent to `for (auto _ : state)` in C++.
    l.index(5000)         # Like C++ only code inside the loop is timed.

if __name__ == '__main__':
  benchmark.main()
```

It doesn't require much to get a simple example like this working. Namely:
1. Have a (private) pybind11 extension to expose a wrapper around:
    - `benchmark::Initialize`
    - `benchmark::RegisterBenchmark`
    - `benchmark::RunSpecifiedBenchmarks`
    - `benchmark::State` (a shim only supporting `KeepRunning` renamed to `__bool__`).
2. Create a Python API exposing:
    - `register` (calls RegisterBenchmark)
    - `main` (calls Initialize, RunSpecifiedBenchmarks)

**Comment by chr1sj0nes** [comment]
Where would be the best place to put such bindings? `//bindings/python/`?

**Comment by dominichamon** [comment]
Great!

I think bindings/python makes sense (or bindings/py if you want to be hip and groovy). This also plays nicely with #776 which is offering fortran/rust bindings.

**Comment by dominichamon** [comment]
Should have resolved this when #957 was merged :)
